packages/protocol/contracts/L1/TaikoL1.sol

```diff
@@ -82,6 +82,13 @@ contract TaikoL1 is
      *          will be the first transaction in the block -- if there are
      *          n transactions in `txList`, then there will be up to n+1
      *          transactions in the L2 block.
+     *
+     *        - inputs[2] a byte32 that determins which verifier to run.
+     *
+     *        - inputs[3] a ZKP that verifies all transactions are valid -- they
```

Comment:
So the reason why we don't verify the "sig zkp" in `proposeBlock` is because we want to reduce the cost of honest proposers? (not sure the actual cost of this new circuits) Since if we do so, then we can totally remove the `block id` concept (like `state.nextBlockId` / `state.latestVerifiedId`) in protocol, and only keep the `block height` concept (like `state.latestVerifiedHeight`) ? and the `proveBlockInvalid` mechanism can also be removed? which may make our protocol more simpler.